libsmtutil/SMTLib2Context.h

```diff
@@ -0,0 +1,119 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+// SPDX-License-Identifier: GPL-3.0
+
+#pragma once
+
+#include <libsmtutil/SolverInterface.h>
+#include <libsmtutil/Sorts.h>
+
+#include <string>
+#include <unordered_map>
+#include <unordered_set>
+
+namespace solidity::smtutil
+{
+
+struct SortId
```

Comment:
Isn't it a bit of an overkill? Maybe `typedef` would do instead?
There would be no `==` operator function then.

Reply:
I generally like strong types, but in this case, you are right, it is an overkill.